### PR TITLE
[NVD] Add package source to agents table

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|#4835|

## Description

For some Linux packages, the NVD database stores the `source` name of the package. In this PR, the `source` name of each agent package is retrieved from syscollector and stored in vulnerability detector `AGENTS` table, so it can be used to check for vulnerabilities.

Also, the NDV database has vulnerabilities related to the OS and kernel version of the Linux systems.  Since syscollector `sys_programs` table doesn't contains this information, a new function has been added to retrieve this infomation from `sys_osinfo` table. Besides, another function creates the OS and kernel packages using the information retrieved from syscollector and stores them in vulnerability detector `AGENTS` table, so vulnerabilities for OS and kernel versions can be analized.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -3732,6 +3762,65 @@ int wm_vuldet_generate_win_cpe(agent_software *agent) {
     return 0;
 }
 
+int wm_vuldet_generate_os_and_kernel_package(sqlite3 *db, agent_software *agent) {
+    static const char *PR_UBUNTU = "ubuntu_linux";
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:3766]: Yes, I agree. Not just these strings, there are many more things that could be moved to a header file (for example, function declarations). I think these strings will be used only in this function, so I think I would leave them here but in the full rework we can reorganize this file. Do you agree?

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -3732,6 +3762,65 @@ int wm_vuldet_generate_win_cpe(agent_software *agent) {
...
+    static const char *PR_DEBIAN = "debian_linux";
+    static const char *PR_REDHAT = "enterprise_linux";
+    static const char *V_UBUNTU = "canonical";
+    static const char *V_DEBIAN = "debian";
+    static const char *V_REDHAT = "redhat";
+    const char *product_name = NULL;
+    const char *vendor = NULL;
+
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:3774]: These are already verified in the `wm_vuldet_check_agent_vulnerabilities` function, but to be completely sure I will verify them here too.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 3]
```diff
@@ -3732,6 +3762,65 @@ int wm_vuldet_generate_win_cpe(agent_software *agent) {
...
+    if (agent->os_release) {
+        if (wm_vuldet_insert_agent_data(db, agent, NULL, vendor, product_name, NULL, agent->os_release, agent->arch, NULL, NULL)) {
+            mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_INSERT_DATA_ERR, product_name);
+            return OS_INVALID;
+        }
+    }
+
+    // Generate kernel package
+    if (agent->kernel_release) {
+        if (wm_vuldet_insert_agent_data(db, agent, NULL, "linux", "linux_kernel", NULL, agent->kernel_release, agent->arch, NULL, NULL)) {
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:3815]: Yes, I will move them to the begining of the function.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 4]
```diff
@@ -4615,6 +4706,73 @@ int wm_vuldet_get_oswindows_info(agent_software *agent) {
     return retval;
 }
 
+int wm_vuldet_get_oslinux_info(agent_software *agent) {
+    int retval = OS_INVALID;
+    char request[OS_SIZE_128];
+    cJSON *obj = NULL;
+    cJSON *obj_it;
+
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:4714]: Yes, will do as explained above.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:3766]: Yes! I agree!

**Pull request merged**